internal/provider/cisco/nxos/provider.go

```diff
@@ -83,6 +79,29 @@ func (p *Provider) Disconnect(_ context.Context, _ *deviceutil.Connection) error
 	return p.conn.Close()
 }
 
+func (p *Provider) HashProvisioningPassword(password string) (hashed, encryptType string, err error) {
+	salt := make([]byte, 10)
+	if _, err := rand.Read(salt); err != nil {
+		return "", "", err
+	}
+	s := [10]byte{}
+	copy(s[:], salt)
```

Comment:
nit: I think we can directly read into the array.

```suggestion
	s := [10]byte{}
	if _, err := rand.Read(s[:]); err != nil {
		return "", "", err
	}
```